test/TestInternalGrains/StreamingImmutabilityTestGrain.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Threading.Tasks;
+using Orleans;
+using Orleans.Streams;
+using UnitTests.GrainInterfaces;
+
+namespace UnitTests.Grains
+{
+    public class StreamingImmutabilityTestGrain : Grain, IStreamingImmutabilityTestGrain
+    {
+        private StreamImmutabilityTestObject _myObject;
+        private StreamSubscriptionHandle<StreamImmutabilityTestObject> _streamSubscriptionHandle;
+
+        public async Task SubscribeToStream(Guid guid, string providerName)
+        {
+            var stream = GetStreamProvider(providerName).GetStream<StreamImmutabilityTestObject>(guid, "Namespace");
+            _streamSubscriptionHandle = await stream.SubscribeAsync(OnNextAsync);
+        }
+
+        public async Task UnsubscribeFromStream()
+        {
+            if (_streamSubscriptionHandle != null)
+                await _streamSubscriptionHandle.UnsubscribeAsync();
+        }
+
+        public async Task ProduceTestObject(string providerName)
+        {
+            _myObject = new StreamImmutabilityTestObject() {MyString = this.GetPrimaryKey().ToString()};
+            var stream = GetStreamProvider(providerName).GetStream<StreamImmutabilityTestObject>(this.GetPrimaryKey(), "Namespace");
+            await stream.OnNextAsync(_myObject);
```

Comment:
> Yes and no. The immutability alters exactly one behavior, weather a deep copy of an object is made at call time. All other 'aspects' are side effects of our RPC pattern.

I think I just fully understood how that immutability is used in the context of multiplexing. Immutability does not make sense for a remote consumer nor should it change anything.
Would you want to change the test similar to what you mentioned in your first comment while differentiating between a local and remote consumer?
Or would you prefer to just test the immutability behavior with a local consumer and not consider the remote consumer case (now my preferred option)?

It's not stickling, hurray for code and test quality :+1: